packages/grid/x-grid/test/XGrid.test.tsx

```diff
@@ -14,13 +27,65 @@ async function sleep(duration: number) {
 describe('<XGrid />', () => {
   const render = createClientRender();
 
-  it('should resize the width of the columns', async function test() {
+  before(function beforeHook() {
     if (/jsdom/.test(window.navigator.userAgent)) {
-      // TODO: Unclear why this fails. Not important
-      // since a browser test gives us more confidence anyway
+      // Need layouting
       this.skip();
     }
+  });
+
+  describe('keyboard', () => {
+    const KeyboardTest = () => {
+      const data = useData(100, 20);
+      const transformColSizes = (columns) => columns.map((column) => ({ ...column, width: 60 }));
+
+      return (
+        <div style={{ width: 300, height: 300 }}>
+          <XGrid rows={data.rows} columns={transformColSizes(data.columns)} />
+        </div>
+      );
+    };
+
+    it('cell navigation with arrows ', async () => {
+      render(<KeyboardTest />);
+      await sleep(10);
+      document.querySelector('[data-rowindex="0"]').querySelector('[data-colindex="0"]').focus();
+      expect(getActiveCell()).to.equal('0-0');
 
+      fireEvent.keyDown(document.activeElement, { code: 'ArrowRight' });
+      await sleep(100);
```

Comment:
Timeout is hardcoded in https://github.com/mui-org/material-ui-x/blob/93550f9c54099f3be06800f8a01a9e09e12a9821/packages/grid/x-grid-modules/src/hooks/root/useKeyboard.ts#L119, we could mock time.

Reply:
yes we should refactor this code and drop the timeout.